src/libolm_migration.rs

```diff
@@ -0,0 +1,440 @@
+        .transpose()?;
+
+    let cross_signing = PrivateCrossSigningIdentity::empty(&user_id);
+    cross_signing
+        .import_secrets_unchecked(
+            data.cross_signing_master_key.as_deref(),
+            data.cross_signing_self_signing_key.as_deref(),
+            data.cross_signing_user_signing_key.as_deref(),
+        )
+        .await?;
+
+    store.save_pending_changes(PendingChanges { account: Some(account) }).await?;
+    store
+        .save_changes(Changes {
+            private_identity: Some(cross_signing),
+            backup_decryption_key,
+            backup_version: data.backup_version.clone(),
+            ..Default::default()
+        })
+        .await?;
+    Ok(())
+}
+
+/// A pickled version of a `Session`.
+///
+/// Holds all the information that needs to be stored in a database to restore
+/// an Olm Session. Can be imported into the rust store with {@link
+/// #migrateOlmSessions}.
+#[wasm_bindgen(getter_with_clone)]
+#[derive(Debug)]
+pub struct PickledSession {
+    /// The pickle string holding the Olm Session, as returned by `olm_pickle_session` in libolm.
+    pub pickle: String,
+    /// The base64-encoded public curve25519 key of the other user that we share
+    /// this session with.
+    #[wasm_bindgen(js_name = "senderKey")]
+    pub sender_key: String,
+    /// Was the session created using a fallback key?
+    #[wasm_bindgen(js_name = "createdUsingFallbackKey")]
+    pub created_using_fallback_key: bool,
+    /// When the session was created.
+    #[wasm_bindgen(js_name = "creationTime")]
+    pub creation_time: Date,
+    /// When the session was last used.
+    #[wasm_bindgen(js_name = "lastUseTime")]
+    pub last_use_time: Date,
+}
+
+#[wasm_bindgen]
+impl PickledSession {
+    /// Construct a new `PickledSession`, with default values.
+    #[wasm_bindgen(constructor)]
+    pub fn new() -> Self {
+        Self::default()
+    }
+}
+
+
+#[wasm_bindgen]
+impl Migration {
+    /// Migrate Olm sessions of a libolm-based setup to a vodozemac-based setup
+    /// stored in an indexedDB crypto store.
+    ///
+    /// Before this method can be used, {@link #migrateBaseData} must be used to
+    /// import the base data into the crypto store.
+    ///
+    /// # Arguments
+    ///
+    /// * `sessions` - An `Array` of {@link PickledSession}s to import.
+    /// * `pickle_key` - The libolm pickle key that was used to pickle the olm
+    ///   session objects.
+    /// * `store_handle` - A connection to the CryptoStore which will be used to
+    ///   store the vodozemac data.
+    #[wasm_bindgen(js_name = "migrateOlmSessions")]
+    pub async fn migrate_olm_sessions(
+        sessions: Array,
+        pickle_key: Uint8Array,
+        store_handle: &StoreHandle,
+    ) -> Result<JsValue, JsError> {
+        let rust_sessions = sessions
+            .into_iter()
+            .map(|s| libolm_pickled_session_to_rust_pickled_session(s, &(pickle_key.to_vec())))
```

Comment:
Just seeing this: it might be possible that `pickle_key.to_vec()` could be hoisted above the `let rust_sessions = ` statement, to avoid creating the pickle key vector for every single session.